sdks/go/pkg/beam/windowing.go

```diff
@@ -21,21 +21,40 @@ import (
 	"github.com/apache/beam/sdks/go/pkg/beam/internal/errors"
 )
 
+type WindowIntoOption interface {
+	windowIntoOption()
+}
+
+type WindowTrigger struct {
+	Name window.TriggerType
+}
+
+func (t WindowTrigger) windowIntoOption() {}
+
 // WindowInto applies the windowing strategy to each element.
-func WindowInto(s Scope, ws *window.Fn, col PCollection) PCollection {
-	return Must(TryWindowInto(s, ws, col))
+func WindowInto(s Scope, ws *window.Fn, col PCollection, opts ...WindowIntoOption) PCollection {
+	return Must(TryWindowInto(s, ws, col, opts...))
 }
 
 // TryWindowInto attempts to insert a WindowInto transform.
-func TryWindowInto(s Scope, ws *window.Fn, col PCollection) (PCollection, error) {
+func TryWindowInto(s Scope, wfn *window.Fn, col PCollection, opts ...WindowIntoOption) (PCollection, error) {
 	if !s.IsValid() {
 		return PCollection{}, errors.New("invalid scope")
 	}
 	if !col.IsValid() {
 		return PCollection{}, errors.New("invalid input pcollection")
 	}
+	ws := window.WindowingStrategy{Fn: wfn, Trigger: window.Default}
+	for _, opt := range opts {
+		switch opt.(type) {
+		case WindowTrigger:
+			ws.Trigger = opt.(WindowTrigger).Name
+		default:
+			ws.Trigger = window.Default
```

Comment:
Since this switch is only run if we have options, it's valid to return an error or even `panic`if we don't know what type of option it is. That way users won't accidently override earlier set triggers because an implementation was half done somehow.